### PR TITLE
Update structure of theming.

### PR DESCRIPTION
**Pull request by nickbutcher**
- Add dark theme colors.
- Add `JetnewsTheme` composable to centrailze theming.
- Hoist dark theme param to enable easy overrides e.g. `@Preview`s
- Add shape theming.

### CODE REVIEW

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt [hunk 1]
```diff
@@ -52,7 +49,10 @@ private fun SelectTopicButtonOn(modifier: Modifier = Modifier) {
         shape = CircleShape,
         modifier = modifier
     ) {
-        Icon(vectorResource(R.drawable.ic_check))
+        Icon(
+            asset = vectorResource(R.drawable.ic_check),
+            tint = MaterialTheme.colors.onPrimary
```
Comment (nickbutcher) [inline JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt:54]: Snuck in a bug fix revealed by dark theme.

File: JetNews/app/src/main/java/com/example/jetnews/ui/theme/Color.kt [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.jetnews.ui.theme
+
+import androidx.ui.graphics.Color
+
+val red200 = Color(0xfff297a2)
```
Comment (Splogy) [inline JetNews/app/src/main/java/com/example/jetnews/ui/theme/Color.kt:21]: Maybe these should be PascalCased? val Red200, etc.

Comment (nickbutcher) [reply JetNews/app/src/main/java/com/example/jetnews/ui/theme/Color.kt:21]: Good call.

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt (hunk 1 above)
Comment (chrisbanes) [reply JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt:54]: Should this have an emphasis?

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt [hunk 3]
```diff
@@ -52,7 +49,10 @@ private fun SelectTopicButtonOn(modifier: Modifier = Modifier) {
         shape = CircleShape,
         modifier = modifier
     ) {
-        Icon(vectorResource(R.drawable.ic_check))
+        Icon(
+            asset = vectorResource(R.drawable.ic_check),
```
Comment (chrisbanes) [inline JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt:53]: Is this available in `Icons`?

Comment (nickbutcher) [reply JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt:53]: Done.

File: JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt (hunk 1 above)
Comment (nickbutcher) [reply JetNews/app/src/main/java/com/example/jetnews/ui/interests/SelectTopicButton.kt:54]: Yeah, good catch. Should be high emphasis. Done.

**Pull request merged**